src/plugins/content_management/common/rpc/search.ts

```diff
@@ -16,25 +16,58 @@ export const searchSchemas: ProcedureSchemas = {
     {
       contentTypeId: schema.string(),
       version: versionSchema,
-      // --> "query" that can be executed will be defined by each content type
-      query: schema.recordOf(schema.string(), schema.any()),
+      query: schema.oneOf([
+        schema.object(
+          {
+            text: schema.maybe(schema.string()),
+            tags: schema.maybe(schema.arrayOf(schema.arrayOf(schema.string()), { maxSize: 2 })),
+            limit: schema.maybe(schema.number()),
+            cursor: schema.maybe(schema.string()),
+          },
+          {
+            unknowns: 'forbid',
+          }
+        ),
+      ]),
       options: schema.maybe(schema.object({}, { unknowns: 'allow' })),
     },
     { unknowns: 'forbid' }
   ),
-  out: schema.oneOf([
-    schema.object({}, { unknowns: 'allow' }),
-    schema.arrayOf(schema.object({}, { unknowns: 'allow' })),
-  ]),
+  out: schema.object({
+    contentTypeId: schema.string(),
+    result: schema.object({
+      hits: schema.arrayOf(schema.any()),
+      pagination: schema.object({
+        total: schema.number(),
+        cursor: schema.maybe(schema.string()),
+      }),
+    }),
+  }),
 };
 
-export interface SearchIn<
-  T extends string = string,
-  Query extends object = object,
-  Options extends object = object
-> {
+export interface SearchQuery {
+  /** The text to search for */
+  text?: string;
+  /** List of tags id to include and exclude */
+  tags?: [included: string[], excluded?: string[]];
```

Comment:
I there a way to only exclude? Should we make API: 


tags?: {
included?: string[],
excluded?: string[]
}

Reply:
Good point. I pushed https://github.com/elastic/kibana/pull/154150/commits/f13eb75cc7f9977c31cbaee933320166c009672b so we can do it like that:

```ts
const query: SearchQuery = {
  tags: [, ['123']],
};
```

Reply:
I still suggest to make it an object.
The downside of the tuple is that it is not readable: 

```
const query: SearchQuery = {
  tags: [, ['123']],
};
```

without drilling down to `SearchQuery` can't say what does `[, ['123']]` mean

Reply:
Valid point. I changed it in https://github.com/elastic/kibana/pull/154150/commits/cb35516c182e4da427115ac43b5f868c9bd5021b